src/cmd/tools/docs_test/main/main.go

```diff
@@ -75,15 +78,48 @@ func main() {
 				link := node.LinkData
 				url := strings.TrimSpace(string(link.Destination))
 
-				parse, ok := docs.ParseRepoPathURL(url)
-				if !ok {
-					break
+				resolvedPath := ""
+				if parse, ok := docs.ParseRepoPathURL(url); ok {
+					// Internal relative local repo path
+					resolvedPath = parse.RepoPath
+				} else if httpOrHTTPS(url) {
+					// External link
+					resolvedPath = url
+				} else {
+					// Otherwise should be a direct relative link
+					// (not repo URL prefixed)
+					if v := strings.Index(url, "#"); v != -1 {
+						// Remove links to subsections of docs
+						url = url[:v]
+					}
+					resolvedPath = path.Join(path.Dir(filePath), url)
 				}
 
-				if _, err := os.Stat(parse.RepoPath); err != nil {
-					return abort(fmt.Errorf(
-						"could not stat repo path: link=%s, path=%s, err=%v",
-						url, parse.RepoPath, err))
+				checking := checkedLink{
+					Document:     filePath,
+					URL:          url,
+					ResolvedPath: resolvedPath,
+				}
+				if httpOrHTTPS(resolvedPath) {
+					// Check external link using HEAD request
+					client := &http.Client{
+						Transport: &http.Transport{
+							// Some sites have invalid certs, like some kubernetes docs
+							TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+						},
+					}
+					resp, err := client.Head(resolvedPath)
+					if err == nil && resp.StatusCode/100 != 2 {
```

Comment:
Haha nice. Would be neat if `http.Response` had like an `isSuccessful` method or something.

Reply:
Yeah lol... sadly not.